sqlq.go

```diff
@@ -168,6 +168,24 @@ retry:
 	return &job, nil
 }
 
+// Cancelled transitions the job to CANCELLED state and mark it as completed
+func Cancelled(cx Connection, job *Job) (n int64, err error) {
+	var ctx = context.Background()
+	var res sql.Result
+
+	const cancelJob = `
```

Comment:
Let's add a `RETURNING *` clause here and use `scanJob()` to update the `job` object here. See `sqlq.Success()` for reference.

Reply:
I just realised that this function is doing multiple things. It is 1) checking for `cancelling` status and 2) moving a job to `cancelled` status. Can we separate these two concerns? maybe add an `IsCancelled()` function?